src/mappings/entities/mapIdentities.ts

```diff
@@ -185,9 +185,16 @@ const handleSecondaryKeysPermissionsUpdated = async (
 ): Promise<void> => {
   const [, rawSignerDetails, , rawUpdatedPermissions] = params;
 
-  const {
-    signer: { account: address },
-  } = JSON.parse(rawSignerDetails.toString());
+  let address;
+  try {
```

Comment:
could this be some kind of check instead of relying on an exception to be thrown? I'd prefer `if` over exception for control flow.

If there is a reason we have to use try / catch its OK, but I think its better to avoid if we can help it

Reply:
I'll look for a work around

Reply:
Updated the logic using instanceof